src/data-access/MwApiLexemeCreator.ts

```diff
@@ -37,10 +45,27 @@ export default class MwApiLexemeCreator implements LexemeCreator {
 				language: lexemeLanguageItemId,
 				lexicalCategory: lexicalCategoryItemId,
 			} ),
+			errorformat: 'html',
+			formatversion: 2,
 		} );
 		const response = await this.api.postWithEditToken( params )
-			.catch( () => {
-				return Promise.reject( [ { type: 'unknown' } ] );
+			// jQuery promises can resolve and reject with multiple values;
+			// the third argument is either missing or the result
+			.catch( ( code: string, _?: unknown, result?: ApiResult ): Promise<never> => {
+				let errors: SubmitError[];
+				try {
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					errors = result!.errors.map( ( apiError ) => {
+						const submitError: SubmitError = { type: apiError.code };
+						if ( apiError.html ) {
+							submitError.message = apiError.html;
+						}
+						return submitError;
+					} );
+				} catch {
+					errors = [ { type: code } ];
+				}
```

Comment:
To make sure I understand correctly, we are unsure is result exists (or if results is an array) before we map it, so we use this `try catch` to pattern match on result? Shouldn't we add more explicit logic to check these conditions? Or am I missing a different scenario where the error mapping would throw?

Reply:
> To make sure I understand correctly, we are unsure if `result` is set (or if it is an array) before we map it, so we use this `try catch` to pattern match on it?

Yes, it might be `undefined` or empty.

> Shouldn't we add more explicit logic to check these conditions?

To me this seems nicer than checking several condition in sequence (is object, has errors, errors are array…).

Reply:
I guess so, though I’d say there’s still a lot going on in that snippet…

Reply:
Yeah, we don't need to check if it's an object, only arrays can be mapped in JS. Anything other than an array will return false for that check. As for the logic inside the callback, that can stay the same, it's just replacing the `try... catch...` (which can inadvertently cause this mapping to fail for other reasons that we might not want to so easily catch) with what the actual intent of this snippet is -  map errors from an array if that array is there and provide us with a code otherwise.

Reply:
(With “is object” I was referring to `result`, not `result.errors`.)

Reply:
We discussed this in a call, I’ve pushed a new commit that shifts to using `if` inside a try/catch (we keep the try/catch to ensure we don’t reject the promise with something other than an array of errors).